### ISSUE TITLE
Google Speaker Group - TTS issues

### ISSUE THREAD

**Issue by jazzyisj** [music-assistant/hass-music-assistant#631]
### What version of Music Assistant has the issue?

2022.7 BETA 3

### The problem

**All media players mentioned are mass media players.**

- When streaming on a Google speaker group and a tts message is played on another Google speaker group that contains members of the media group, the tts message plays but the media stream does not resume when tts has completed.

- When streaming on a Google speaker group and a tts message is played a **single** media group member the tts message plays on **all** group members.

- When streaming on a single Google speaker and a tts message is played on a Google speaker group that contains the media stream player, the media stream does not resume after tts.

### How to reproduce

Play media and TTS calls as described above.

### Relevant log output

None.

### What version of Home Assistant Core are your running

2022.7.2

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

ODROID

**Comment by marcelveldt** [comment]
Just tested it and I can confirm the TTS/announce is flaky, especially with cast groups somehow. I think this relates to the recent buffer changes. I will have a look.

**Comment by marcelveldt** [comment]
1) Can you please check if this is fixed in the 2022.7 beta 4 ?


2) This is intended behavior. If the group is playing, that is the active "queue" and you can't send direct commands to the childs without first stopping the group.


3) Can you please check if this is fixed in the 2022.7 beta 4 ?

**Comment by erkr** [comment]
@jazzyisj 
Since 2022.7b7, TTS seems to work robust for the simple use cases. Please verify yours if you have time before we come this issue. Best Eric

**Comment by jazzyisj** [comment]
Oh shoot sorry guys!  Missed this somehow.

This still doesn't seem to be resolved for me.  Updated v2022.7b7, HASS is v2022.7.5

```
2022-07-16 16:13:02 WARNING (MainThread) [homeassistant.helpers.entity] Update of binary_sensor.wan_connected is taking over 10 seconds
2022-07-16 16:13:04 WARNING (SyncWorker_32) [pychromecast.controllers] Quick Play failed for http://192.168.1.100:8095/219b5510b0ce424a937017effb451a42.flac:audio/flac({'enqueue': False, 'stream_type': 'LIVE', 'title': ' Streaming from Music Assistant'})
2022-07-16 16:13:07 WARNING (MainThread) [homeassistant.components.media_player] Updating webostv media_player took longer than the scheduled update interval 0:00:10
2022-07-16 16:13:07 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 451, in resume
    await self.play_index(resume_item.item_id, resume_pos, fade_in)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 526, in play_index
    await self.player.play_url(stream.url)
  File "/config/custom_components/mass/player_controls.py", line 525, in play_url
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-07-16 16:13:08 WARNING (MainThread) [music_assistant.players.stream] Abort: client(s) did not connect within 10 seconds.
```
Playing on a single cast speaker, then TTS on google group speaker - no resume.
```
2022-07-16 16:16:27 ERROR (MainThread) [music_assistant.players] Error while playing announcement
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 365, in play_announcement
    await self._wait_for_state(PlayerState.PLAYING, silence_item.item_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 867, in _wait_for_state
    raise TimeoutError(f"Timeout while waiting on state(s) {state}")
TimeoutError: Timeout while waiting on state(s) (,)
2022-07-16 16:17:05 WARNING (SyncWorker_21) [pychromecast.controllers] Quick Play failed for http://192.168.1.100:8095/880f698a310f46588e9c3a3a8f66874e.flac:audio/flac({'enqueue': False, 'stream_type': 'LIVE', 'title': ' Streaming from Music Assistant'})
2022-07-16 16:17:05 WARNING (MainThread) [music_assistant.players.stream] Abort: client(s) did not connect within 10 seconds.
2022-07-16 16:17:05 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall media_player.play_media (c:01G84AG81V11CYJW4KZF5GBNSS): entity_id=['media_player.mass_dining_room_hub'], extra=, media_type=audio/flac, media_id=media-source://mass/database://playlist/31>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1731, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1750, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 930, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 717, in _handle_entity_call
    await result
  File "/config/custom_components/mass/media_player.py", line 427, in async_play_media
    await self.player.active_queue.play_media(media_id, queue_opt)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 272, in play_media
    await self.load(queue_items, passive)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 568, in load
    await self.play_index(0, passive=passive)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 526, in play_index
    await self.player.play_url(stream.url)
  File "/config/custom_components/mass/player_controls.py", line 525, in play_url
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-07-16 16:17:41 WARNING (SyncWorker_24) [pychromecast.controllers] Quick Play failed for http://192.168.1.100:8095/f50aba92e80b4146bf1d4eb741665c6d.flac:audio/flac({'enqueue': False, 'stream_type': 'LIVE', 'title': ' Streaming from Music Assistant'})
2022-07-16 16:17:43 WARNING (MainThread) [music_assistant.players.stream] Abort: client(s) did not connect within 10 seconds.

```

**Comment by erkr** [comment]
@jazzyisj : thanks for your additional info! Marcel included this issue on the ambition list for the next official release 2022.7.0

**Comment by marcelveldt** [comment]
I can't reproduce any of this with current version (2022.7 beta 10) so I'm assuming/hoping it is fixed.

**Comment by marcelveldt** [comment]
Please reopen (or create a new issue) if you still may experience issues. Thanks!
